test/public/logs/overview.test.js

```diff
@@ -138,20 +142,19 @@ module.exports = () => {
 
         // Open the filters
         await pressElement(page, '#openFilterToggle');
-        await waitForTimeout(20);
 
         // Insert some text into the filter
-        await page.type('#contentFilterText', 'particle');
-        await waitForTimeout(300);
+        await fillInput(page, '#contentFilterText', 'particle');
+        await page.waitForFunction(() => document.querySelectorAll('table tbody tr').length === 2, { timeout: 1500 });
 
         // Expect the new total number of rows to be less than the original number of rows
         const firstFilteredRows = await page.$$('table tr');
         const firstFilteredNumberOfRows = firstFilteredRows.length - 1;
         expect(firstFilteredNumberOfRows).to.be.lessThan(originalNumberOfRows);
 
         // Insert some other text into the filter
-        await page.type('#titleFilterText', 'this-content-do-not-exists-anywhere');
-        await waitForTimeout(300);
+        await fillInput(page, '#titleFilterText', 'this-content-do-not-exists-anywhere');
+        await page.waitForFunction(() => document.querySelectorAll('table tbody tr').length === 1, { timeout: 1500 });
```

Comment:
Be very careful with this: loading table and empty table will match this selector.

Reply:
Done, used waitForTableToLength with expectedSize as 0